pkg/git/fetch.go

```diff
@@ -22,60 +16,55 @@ type Fetcher struct {
 }
 
 func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
-	resolvedAuth, err := f.Keychain.Resolve(gitURL)
-	if err != nil {
-		return err
-	}
-
-	tmpDir, err := ioutil.TempDir("", "git-clone-")
-	if err != nil {
-		return err
-	}
+	f.Logger.Printf("Cloning %q @ %q...", gitURL, gitRevision)
 
-	repo, err := git.PlainInit(tmpDir, false)
+	repository, err := git2go.InitRepository(dir, false)
 	if err != nil {
-		return errors.Wrap(err, "unable to init git repository")
+		return errors.Wrap(err, "initializing repo")
 	}
+	defer repository.Free()
 
-	remote, err := repo.CreateRemote(&config.RemoteConfig{
-		Name: "origin",
-		URLs: []string{gitURL},
+	remote, err := repository.Remotes.CreateWithOptions(gitURL, &git2go.RemoteCreateOptions{
+		Name:  "origin",
+		Flags: git2go.RemoteCreateSkipInsteadof,
```

Comment:
This is for local to avoid your local git config instead of overwriting the repo url.